src/dart.rs

```diff
@@ -35,23 +35,37 @@ impl DartGenerator {
             import "dart:isolate";
             import "dart:typed_data";
 
-            ffi.Pointer<ffi.Void> _registerFinalizer(_Box boxed) {
-                final dart = ffi.DynamicLibrary.executable();
-                final registerPtr = dart.lookup<ffi.NativeFunction<ffi.Pointer<ffi.Void> Function(
-                    ffi.Handle, ffi.Pointer<ffi.Void>, ffi.IntPtr, ffi.Pointer<ffi.Void>)>>("Dart_NewFinalizableHandle");
-                final register = registerPtr
-                    .asFunction<ffi.Pointer<ffi.Void> Function(
-                        Object, ffi.Pointer<ffi.Void>, int, ffi.Pointer<ffi.Void>)>();
-                return register(boxed, boxed._ptr, 42, boxed._dropPtr.cast());
+            class _DartApiEntry extends ffi.Struct {
+                external ffi.Pointer<ffi.Uint8> name;
+                external ffi.Pointer<ffi.Void> ptr;
             }
 
-            void _unregisterFinalizer(_Box boxed) {
-                final dart = ffi.DynamicLibrary.executable();
-                final unregisterPtr = dart.lookup<ffi.NativeFunction<ffi.Void Function(
-                    ffi.Pointer<ffi.Void>, ffi.Handle)>>("Dart_DeleteFinalizableHandle");
-                final unregister = unregisterPtr
-                    .asFunction<void Function(ffi.Pointer<ffi.Void>, _Box)>();
-                unregister(boxed._finalizer, boxed);
+            class _DartApi extends ffi.Struct {
+                @ffi.Int32()
+                external int major;
+
+                @ffi.Int32()
+                external int minor;
+
+                external ffi.Pointer<_DartApiEntry> functions;
+            }
+
+            ffi.Pointer<T> _lookupDartSymbol<T extends ffi.NativeType>(String symbol) {
+                final ffi.Pointer<_DartApi> api = ffi.NativeApi.initializeApiDLData.cast();
+                final ffi.Pointer<_DartApiEntry> functions = api.ref.functions;
+                for (var i = 0; i < 100; i++) {
+                    final func = functions.elementAt(i).ref;
+                    var symbol2 = "";
+                    var j = 0;
+                    while (func.name.elementAt(j).value != 0) {
+                        symbol2 += String.fromCharCode(func.name.elementAt(j).value);
+                        j += 1;
+                    }
+                    if (symbol == symbol2) {
+                        return func.ptr.cast();
+                    }
+                }
+                throw "symbol not found";
```

Comment:
Can't you pre-fill a map or something?

Reply:
we only use it to locate two symbols at startup, which are then cached, so it's probably not worth it